TypeSystem/Evolution/type_evolution.h

```diff
@@ -61,6 +68,32 @@ struct Evolve;
 #include "../primitive_types.dsl.h"
 #undef CURRENT_DECLARE_PRIMITIVE_TYPE
 
+// Boilerplate default generic evolutor for `std::vector<T>`.
+template <typename FROM_NAMESPACE, typename EVOLUTOR, typename VECTOR_ELEMENT_TYPE>
+struct Evolve<FROM_NAMESPACE, std::vector<VECTOR_ELEMENT_TYPE>, EVOLUTOR> {
+  template <typename INTO, typename OUTPUT>
+  static void Go(const std::vector<VECTOR_ELEMENT_TYPE>& from, OUTPUT& into) {
+    into.resize(from.size());
```

Comment:
Our `Evolve<...>::Go(from, into)` requires an existing placeholder, and `.resize()` allocates it and makes it accessible. The alternative would be to `return` the evolved objects and `emplace_back` them, but we decided `OUTPUT& output` as the 2nd parameter to `Evolve<...>::Go()` is a cleaner way